File: alligator/src/main/java/me/aartikov/alligator/animations/providers/DefaultTransitionAnimationProvider.java
package me.aartikov.alligator.animations.providers;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import me.aartikov.alligator.Screen;
import me.aartikov.alligator.TransitionType;
import me.aartikov.alligator.animations.AnimationData;
import me.aartikov.alligator.animations.TransitionAnimation;

/**
 * Date: 24.02.2017
 * Time: 19:14
 *
 * @author Artur Artikov
 */

/**
 * Default implementation of {@link TransitionAnimationProvider}. Always returns {@code TransitionAnimation.DEFAULT}.
 */
public class DefaultTransitionAnimationProvider implements TransitionAnimationProvider {

	@Override
	@NonNull
	public TransitionAnimation getAnimation(@NonNull TransitionType transitionType, @NonNull Class<? extends Screen> screenClassFrom, @NonNull Class<? extends Screen> screenClassTo,
	                                        boolean isActivity, @Nullable AnimationData animationData) {
		return TransitionAnimation.DEFAULT;
	}
}

File: alligator/src/main/java/me/aartikov/alligator/animations/providers/DefaultTransitionAnimationProvider.kt
package me.aartikov.alligator.animations.providers

import me.aartikov.alligator.DestinationType
import me.aartikov.alligator.Screen
import me.aartikov.alligator.TransitionType
import me.aartikov.alligator.animations.AnimationData
import me.aartikov.alligator.animations.TransitionAnimation

/**
 * Default implementation of [TransitionAnimationProvider]. Always returns `TransitionAnimation.DEFAULT`.
 */
class DefaultTransitionAnimationProvider : TransitionAnimationProvider {
    override fun getAnimation(
        transitionType: TransitionType,
        destinationType: DestinationType,
        screenClassFrom: Class<out Screen>,
        screenClassTo: Class<out Screen>,
        animationData: AnimationData?
    ): TransitionAnimation {
        return TransitionAnimation.DEFAULT
    }
}